src/components/DayPickerRangeController.jsx

```diff
@@ -413,36 +418,49 @@ export default class DayPickerRangeController extends React.Component {
   }
 
   onDayClick(day, e) {
-    const { keepOpenOnDateSelect, minimumNights, onBlur } = this.props;
+    const {
+      keepOpenOnDateSelect, minimumNights, onBlur, range,
+    } = this.props;
     if (e) e.preventDefault();
     if (this.isBlocked(day)) return;
 
     const { focusedInput, onFocusChange, onClose } = this.props;
     let { startDate, endDate } = this.props;
 
-    if (focusedInput === START_DATE) {
-      onFocusChange(END_DATE);
+    if (range) {
+      startDate = range.before ? range.before(day.clone()) : day.clone();
```

Comment:
there's 4 clones here; are all 4 needed?

Reply:
They're not, I'll clean this up.